### PR TITLE
feat(toasts): MVP - No Animation

### PR DESCRIPTION
**Pull request by FMorrison87**
## Brief Description

This PR is for the stripped down, basic `rux-toast` and `rux-toast-stack` components. No animation, no toast amount restriction in the toast stack.

A toast is a trivial alert that pops up within the user's field-of-view in any of the corners of the viewport. Toasts stack and array downward as they come in.

The `rux-toast` props are `message`, `hideClose` and `closeAfter` which handles the toast message, hides the close icon, and adds in auto-close functionality, respectively.

The `rux-toast-stack` component manages the placement and imperative creation of toasts. It includes the `position` prop which handles where the stack is placed in the viewport and the `addToast()` method which handles creating toasts and adding them to the particular stack.

## JIRA Link

[TOAST-MVP Ticket](https://rocketcom.atlassian.net/browse/ASTRO-6351)

## General Notes

This component is meant to be a relatively bare bones, but highly extensible MVP for a toast alert. We can, and should extend it to have toast animation, and toast stack restriction functionality. Animation as it is a better experience for the app user so toasts appearing in the viewport is not so jarring, and stack restriction as design has stipulated that we don't want to have more than 4 toasts displaying at a time in the stack. For stack restriction, not having this set at the component level does not restrict the app developer from adding their own styling.

There are two separate branches where most of the animation and a good deal of the stack restriction work has been completed, but it is well behind this MVP branch, so it will need to be carefully merged if that work is to be carried over:
**fm.6189-poc-toasts**
**fm.toast-MVP**

## Motivation and Context

These new components solve the problem of app developers wanting an app alert for tangential notifications. Toasts should be trivial enough that if they are not seen or read, they do not affect critical user action, but they serve the purpose of lower-level confirmations, such as for save states.

## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Checklist

- [x] This PR adds or removes a Storybook story.
- [x] I have added tests to cover my changes.
- [x] Regressions are passing and/or failures are documented
- [x] Changes have been checked in evergreen browsers

### CODE REVIEW

File: packages/web-components/src/components/rux-toast/rux-toast.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        this._handleSlotChange = this._handleSlotChange.bind(this)
+        this._updated()
+        this.hasMessageSlot = hasSlot(this.el)
+    }
+
+    componentDidLoad() {
+        this._handeleOpen()
+    }
+
+    private _handeleOpen() {
```
Comment (markacianfrani) [inline packages/web-components/src/components/rux-toast/rux-toast.tsx:84]:
```suggestion
    private _handleOpen() {
```

File: packages/web-components/src/components/rux-toast/rux-toast.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    }
+
+    connectedCallback() {
+        this._handleSlotChange = this._handleSlotChange.bind(this)
+        this._updated()
+        this.hasMessageSlot = hasSlot(this.el)
+    }
+
+    componentDidLoad() {
+        this._handeleOpen()
```
Comment (markacianfrani) [inline packages/web-components/src/components/rux-toast/rux-toast.tsx:81]:
```suggestion
        this._handleOpen()
```

File: packages/web-components/src/stories/toast-stack.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            position: 'top-right',
+        }}
+    >
+        {Default.bind()}
+    </Story>
+</Canvas>
+
+## Usage
+
+The toast stack component can create toasts via internal method `addToast()`. Toasts created via this method can have their props set via an object passed into the method, and will be automaticaly added to the stack that they were created from.
```
Comment (micahjones13) [inline packages/web-components/src/stories/toast-stack.stories.mdx:49]:
```suggestion
The toast stack component can create toasts via the internal method `addToast()`. Toasts created via this method can have their props set via an object passed into the method, and will be automatically added to the stack that they were created from.
```

File: packages/web-components/src/stories/toast.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+# Toast
+
+Alerts that confirm user actions, and typically don’t require a response, may be displayed in a toast. Toasts should display within the toast-stack component, in the user’s field-of-view. A toast will overlay ui content, but shouldn't block higher level alerts, such as those in a monitoring icon in the Global Status Bar.
+
+A toast should not contain critical information or require user action.
+
+Toasts should array down in a toast stack, with the most recent toast appearing and slotting in at the top.
+
+Toasts can be created imperatively via method using the toast-stack component. A toast on its own will need to be added into the appropriate toast stack.
```
Comment (micahjones13) [inline packages/web-components/src/stories/toast.stories.mdx:33]:
```suggestion
Toasts can be created imperatively via the `addToast()` method in the toast-stack component. A toast on its own will need to be added into the appropriate toast stack.
```

Comment (FMorrison87) [reply packages/web-components/src/stories/toast.stories.mdx]: Fixed!

File: packages/web-components/src/stories/toast-stack.stories.mdx (hunk 3 above)
Comment (FMorrison87) [reply packages/web-components/src/stories/toast-stack.stories.mdx]: Fixed!

File: packages/web-components/src/components/rux-toast-stack/rux-toast-stack.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        const toasts = this._toastsArray
+
+        if (toasts) {
+            for (const [index, value] of toasts?.entries()) {
+                index === 0
+                    ? value.setAttribute('first-toast', '')
+                    : value.removeAttribute('first-toast')
+            }
+        }
+    }
```
Comment (micahjones13) [inline packages/web-components/src/components/rux-toast-stack/rux-toast-stack.tsx:73]: What's the need to tag the first toast? I don't see the output of this function being used anywhere (ie, why do you need to set `first-toast`)

**Review by micahjones13 — APPROVED** [review]

File: packages/web-components/vrt-snaps/rux-dialog/test/basic/dialog.vrt.spec.ts-snapshots/Dialog-focus-state-has-no-visual-regression-vrt-1-chromium-vrt-dark-linux.png
Comment (markacianfrani) [inline packages/web-components/vrt-snaps/rux-dialog/test/basic/dialog.vrt.spec.ts-snapshots/Dialog-focus-state-has-no-visual-regression-vrt-1-chromium-vrt-dark-linux.png:1]: can you remove these dialog VRTs? theres some flake with these that we need to look at separately from this. they shouldntve updated

File: packages/web-components/src/stories/toast.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,205 @@
+import { extractArgTypes } from '@astrouxds/storybook-addon-docs-stencil'
+import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs'
+import { html, render } from 'lit-html'
+const args = extractArgTypes('rux-toast')
+args.message.defaultValue =
+    'This is a toast.'
+
+<Meta
+    title="Components/Toast Stack/Toast"
```
Comment (markacianfrani) [inline packages/web-components/src/stories/toast.stories.mdx:9]: can you copy all the stuff from Timeline about Beta and move it over here? the beta/styleddiv and move the story under the Beta level?

**Review by markacianfrani — APPROVED** [review]

File: packages/web-components/vrt-snaps/rux-dialog/test/basic/dialog.vrt.spec.ts-snapshots/Dialog-focus-state-has-no-visual-regression-vrt-1-chromium-vrt-dark-linux.png
Comment (FMorrison87) [reply packages/web-components/vrt-snaps/rux-dialog/test/basic/dialog.vrt.spec.ts-snapshots/Dialog-focus-state-has-no-visual-regression-vrt-1-chromium-vrt-dark-linux.png:1]: All set.

**Pull request merged**